### PR TITLE
Don't use parameters to pass secrets to GCP or AWS

### PR DESCRIPTION
**Pull request by spikecurtis**
First of several PRs for #1967 

This updates the AWS and GCP templates.  I'd like to also do Kubernetes and Digital Ocean, but I'll need to chase down credentials and/or a test cluster, so let's not block this PR on that.

### CODE REVIEW

File: docs/templates.md [hunk 1]
```diff
@@ -53,6 +51,42 @@ Templates often contain *parameters*. In Coder, there are two types of parameter
...
+
+Additional providers may be supported; check the
+[documentation of the Terraform provider](https://registry.terraform.io/browse/providers) for
+details.
+
+The way these generally work is via the credentials being available to Coder either in some
+well-known location on disk (e.g. `~/.aws/credentials` for AWS on posix systems), or via
+environment variables.  It is usually sufficient to authenticate using the CLI or SDK for the
+cloud provider before running Coder for this to work, but check the Terraform provider
+documentation for details.
```
Comment (bpmct) [inline docs/templates.md:88]: Appreciate the detail here 👍🏼

File: examples/templates/aws-linux/README.md [hunk 2]
```diff
@@ -10,6 +10,13 @@ tags: [cloud, aws]
 
 Pick this template in `coder templates init` and follow instructions.
 
+## Authentication
+
+This template assumes that coderd is run in an environment that is authenticated
+with AWS. For example, run `aws configure import` to import credentials on the
+system and user running coderd.  For other ways to authenticate [consult the
+Terraform docs](https://registry.terraform.io/providers/hashicorp/aws/latest/docs#authentication-and-configuration).
```
Comment (bpmct) [inline examples/templates/aws-linux/README.md:18]: Currently, this isn't really be visible when creating a template via `coder templates init`. I added a mock "variable" in the Docker example, but I'm not really happy with this.

Any ideas? Perhaps it's also fair to rely on the default Terraform errors (+ these docs) to warn users.

**Review by bpmct — APPROVED** [review]
I think this will simplify things a lot, thanks.

File: examples/templates/aws-linux/README.md (hunk 2 above)
Comment (spikecurtis) [reply examples/templates/aws-linux/README.md:18]: Yeah, I don't like the idea of hijacking parameters to guide end users.  If we can't expect users to read the README in the examples, and want do do some handholding, we should build out a guided `template init`

**Pull request merged**